src/stream.jl

```diff
@@ -314,7 +314,7 @@ end
 # Read Functions
 # --------------
 
-function Base.read(stream::TranscodingStream, ::Type{UInt8})
+function Base.peek(stream::TranscodingStream, ::Type{UInt8})::UInt8
```

Comment:
Could you add a comment here so that someone in the future understands how this is related to `peek(::TranscodingStream, ::Type{Char})` ?